clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -464,6 +481,12 @@ public static class LogOffsetPosition {
         public final int position;
         public final int size;
 
+        public LogOffsetPosition(FileChannelRecordBatch batch) {
```

Comment:
Since this LogOffsetPosition is used like a value class, this might be better as an factory method like `LogOffsetPosition#fromBatch` instead of an alternate constructor